base/regex.jl

```diff
@@ -303,12 +303,14 @@ function _write_capture(io, re, group)
     io.size = max(io.size, io.ptr - 1)
 end
 
+const SUB_CHAR = '\\'
```

Comment:
Why does this need to be allowed? That's not covered by tests AFAICT.

Reply:
I added tests for this @nalimilan.

Reply:
Thanks, but I still don't get why backslashes need to be in the list of accepted chars.

Reply:
When the ` _replace function` (in `regex.jl`) sees a backslash, it expects another backslash, a digit or a 'g', and throws an error with `replace_err` if it doesn't see one of those.